src/internal/bolt/bolt-protocol-v1.js

```diff
@@ -262,16 +277,12 @@ export default class BoltProtocol {
     })
 
     // bookmark and mode are ignored in this version of the protocol
-    assertTxConfigIsEmpty(txConfig, this._connection, observer)
+    assertTxConfigIsEmpty(txConfig, this._onProtocolError, observer)
     // passing in a database name on this protocol version throws an error
-    assertDatabaseIsEmpty(database, this._connection, observer)
+    assertDatabaseIsEmpty(database, this._onProtocolError, observer)
 
-    this._connection.write(
-      RequestMessage.run(query, parameters),
-      observer,
-      false
-    )
-    this._connection.write(RequestMessage.pullAll(), observer, flush)
+    this.write(RequestMessage.run(query, parameters), observer, false)
+    this.write(RequestMessage.pullAll(), observer, flush)
```

Comment:
I guess this is the same as the old code but the observer will be completed trice, once for run and once for pullAll. Would it be simpler in the impl of observer to receive only one completed (but errors from both)? Then you can create a temp observer for run message.

Reply:
It receive both, the `ResultStreamObserver` is stateful and move the internal state machine in accordance with the events received.

Reply:
Just meant that it is a bit "leaky" to the observer that pipelining is in use or that more than one command is needed on bolt level at all. But maybe ResultStreamObserver kicks off some other event when RUN is acked